### DISCUSSION TITLE
Couldn't find actually running Talos image name

### DISCUSSION THREAD

**Discussion by marsaltamas0** [siderolabs/talos#9111] [General]
Hi All,

Is there any way to check what is the currently loaded/running Talos image name (with tag)?
E.g.:
**installer:v1.7.4-1-ggb76755-custom**

After a Talos upgrade, machineconfig only contains the old image name.

talosctl version prints out the version only:
E.g.:
**v1.7.4**

Can I get somehow the full image name of the loaded Talos image?

Thanks in advance!

Br,
Tamás

**Comment by smira** [comment]
No, this is not possible unfortunately, as the image might be tagged/copied without changing the contents.

**Reply by marsaltamas0** [reply]
Thank You for your quick answer!

Is there any functionality/feature plan for this in the future?

**Reply by smira** [reply]
Not in the way you ask about it.

Talos version is available via `talosctl version`. If installed from Image Factory, schematic ID is visible in `talosctl get extensions`. This is enough to build an installer image reference for an upgrade.

**Reply by marsaltamas0** [reply]
I see.

Thank You anyway!

**Reply by stereobutter** [reply]
I'd also like to be able to query which specific image is currently installed on a node (to then automatically figure out what update to apply). Couldn't installs/updates simply save the image-ref that was request/pulled similar to `machine.install.image`?

**Reply by stereobutter** [reply]
I've had a look at the code that takes an image ref and then performs the upgrade

https://github.com/siderolabs/talos/blob/6bd6c9b5a08ca3b0e9574e1a61edc54c6ff722bb/internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer_tasks.go#L1290-L1330

Basically the image reference is passed into `install.RunInstallerContainer` which in the end calls out to `/bin/installer`. Could not for example `installer` write out the image reference into a file during installation?. A controller could then pick up that file and read its content into a resource.

**Reply by smira** [reply]
In general - no, Talos might be running in memory while the install is running, so there is no place to save it.

See above - Talos reports its own version and schematic ID, that is enough to reconstruct the image factory URL, and decide if the upgrade is needed. Everything else is local to your environment if you're using non-official images.

**Reply by stereobutter** [reply]
Ah got it. I assume image factory basically bakes the schematic ID into the resulting image and I'd have to do the same for my locally build images (using `imager`)?

**Reply by smira** [reply]
For your local images, you can use something else to figure this out, but Image Factory generates an empty extension with schematic ID which is baked into the image just to populate metadata in `talosctl get extensions`.

**Reply by stereobutter** [reply]
The empty extension is a clever solution, thanks for the tip :)

**Comment by stereobutter** [comment]
Somewhat related to https://github.com/siderolabs/talos/issues/10295
